apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -522,13 +522,13 @@ def close_transport(self):
         self._serial = None
 
     def read_transport(self, n, timeout_sec):
-        # It's hard to set timeout_sec, so we just throw it away
-        # TODO fix this
+        self._serial.timeout = timeout_sec
         if self._serial is None:
             raise server.TransportClosedError()
         return self._serial.read(n)
 
     def write_transport(self, data, timeout_sec):
+        self._serial.timeout = timeout_sec
```

Comment:
i think you should be setting write_timeout, right?

Reply:
Nice catch!